mutable_batch/src/payload/partition.rs

```diff
@@ -49,6 +51,14 @@ pub fn partition_batch<'a>(
     batch: &'a MutableBatch,
     template: &'a TablePartitionTemplateOverride,
 ) -> impl Iterator<Item = (Result<String, PartitionKeyError>, Range<usize>)> + 'a {
+    let parts = template.len();
+    if parts > MAXIMUM_NUMBER_OF_TEMPLATE_PARTS {
+        panic!(
+            "partition template contains {}, which exceeds the maximum of {}",
```

Comment:
For clarity for the person theoretically getting paged over this, I'd change the message to include the word "parts" so that someone doesn't start wondering why a literal 9 isn't allowed in a partition template:

```suggestion
            "partition template contains {} parts, which exceeds the maximum of {} parts",
```